pyvista/core/datasetattributes.py

```diff
@@ -711,6 +711,24 @@ def _prepare_array(
         if data is None:
             raise TypeError('``data`` cannot be None.')
 
+        # convert to numpy type if necessary
```

Comment:
Thanks for the change. Could you please comment on your explanation in the PullRequest summary so that later reviewers of this PullRequest will understand the reason why you moved this code here?

Reply:
I edited the initial top-level comment and added the reason there. Is that the right place to add this info?

Reply:
Yes. Thanks a lot.